contracts/utils/interfaces/IExtendedMulticall.sol

```diff
@@ -13,4 +13,20 @@ interface IExtendedMulticall is IMulticall {
     function getBlockTimestamp() external view returns (uint256);
 
     function getBlockBasefee() external view returns (uint256);
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    event FailedDelegatedcall(
+        bytes data,
+        uint256 timestamp,
+        string errorMessage,
+        address indexed sender
+    );
```

Comment:
I really dislike this duplication but since `ExtendedMulticall` doesn't implement the `IExtendedMulticall` interface I don't see a way around it. I realized that making the contract implement it makes compilation fail due to `IMulticall` and `Multicall` being separated. I'll see if I can find a way around it